src/DssSpell.t.sol

```diff
@@ -651,4 +650,333 @@ contract DssSpellTest is DssSpellTestBase {
+        rwaurn_007.wipe(daiToPay);
+        rwaurn_007.free(1 * WAD);
+
+        // check if we get back RWA009 Tokens
+        assertEq(rwagem_007.balanceOf(address(this)), 1 * WAD, "RWA007: gem not sent back to the caller");
+
+        // check if we have 0 collateral and outstanding debt in the VAT
+        (ink, art) = vat.urns("RWA007-A", address(rwaurn_007));
+        assertEq(ink, 0, "RWA007: bad `ink` after free()");
+        assertEq(art, 0, "RWA007: bad `art` after wipe()");
+    }
+
+    function testFailRWA007_DRAW_ABOVE_LINE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 2_000_000 * WAD;
+
+        // setting address(this) as operator
+        hevm.store(address(rwaurn_007), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+
+        // Draw 2mm
+        rwaurn_007.draw(drawAmount);
+    }
+
+    function testRWA007_OPERATOR_LOCK_DRAW_CAGE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 1_000_000 * WAD;
+
+        // setting address(this) as operator
+        hevm.store(address(rwaurn_007), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+        assertEq(rwaurn_007.can(address(this)), 1);
+
+        // Check if spell lock 1 * WAD of RWA009
+        assertEq(rwagem_007.balanceOf(addr.addr('MCD_PAUSE_PROXY')), 0, "RWA007: gem not transfered from the pause proxy");
+        assertEq(rwagem_007.balanceOf(address(rwajoin_007)), 1 * WAD, "RWA007: gem not locked into the urn");
+
+        // 0 DAI in Output Conduit
+        assertEq(dai.balanceOf(address(rwaconduitout_007)), 0, "RWA007: Dangling Dai in input conduit before draw()");
+
+        // Draw 1mm
+        rwaurn_007.draw(drawAmount);
+
+        // 1mm DAI in Output Conduit
+        assertEq(dai.balanceOf(address(rwaconduitout_007)), drawAmount, "RWA007: Dai drawn was not send to the recipient");
+
+        (uint256 ink, uint256 art) = vat.urns("RWA007-A", address(rwaurn_007));
+        assertEq(art, drawAmount, "RWA007: bad `art` after spell"); // DAI drawn == art as rate should always be 1 RAY
+        assertEq(ink, 1 * WAD, "RWA007: bad `ink` after spell"); // Whole unit of collateral is locked
+
+        hevm.warp(block.timestamp + 10 days);
+        jug.drip("RWA007-A");
+
+        (, uint256 rate,,,) = vat.ilks("RWA007-A");
+        assertEq(rate, RAY, 'RWA007: bad `rate`'); // rate keeps being 1 RAY
+
+        // wards
+        giveAuth(address(rwaconduitout_007), address(this));
+        // may
+        rwaconduitout_007.mate(address(this));
+        rwaconduitout_007.hope(address(this));
+
+        rwaconduitout_007.kiss(address(this));
+        assertEq(rwaconduitout_007.bud(address(this)), 1);
+        rwaconduitout_007.pick(address(this));
+
+        uint256 pushAmount = 100 * WAD; // We push only 100 DAI
```

Comment:
@hbarcelos is there a reason not to push the whole amount ?

```suggestion
        uint256 pushAmount = 1_000_000 * WAD;
```

Reply:
On Görli it does fail, I assume due to lack of liquidity in the PSM, but on mainnet it seems fine

Reply:
I think for mainnnet we can push the whole amount,the issue was with PSM on Goerli which doesn't have enough liquidity to swap 1m to USDC